### PR TITLE
Enforce compiling the library for a certain Java version

### PR DESCRIPTION
**Pull request by matozoid**
As discussed in https://github.com/javaparser/javaparser/issues/100

### CODE REVIEW

**Comment by matozoid** [comment]
Aaaand I still need to fix something here.

File: javaparser-core/pom.xml [hunk 1]
```diff
@@ -28,6 +33,61 @@
...
+                        <id>enforce-versions</id>
+                        <phase>verify</phase>
+                        <goals>
+                            <goal>enforce</goal>
+                        </goals>
+                        <configuration>
+                            <rules>
+                                <requireJavaVersion>
+                                    <!-- Make sure a compiler of this version is used -->
+                                    <version>1.6</version>
```
Comment (ftomassetti) [inline javaparser-core/pom.xml]: Can we refer to the property declared above?

File: javaparser-core/pom.xml [hunk 2]
```diff
@@ -28,6 +33,61 @@
...
+                                </requireJavaVersion>
+                                <enforceBytecodeVersion>
+                                    <!-- Make sure the dependencies are compiled for our Java version -->
+                                    <maxJdkVersion>1.6</maxJdkVersion>
```
Comment (ftomassetti) [inline javaparser-core/pom.xml]: As above, can we refer to the property java.version?

File: javaparser-testing/src/test/java/com/github/javaparser/bdd/ManipulationTest.java [hunk 3]
```diff
@@ -15,7 +15,7 @@
 
     @Override
     public InjectableStepsFactory stepsFactory() {
-        Map<String, Object> state = new HashMap<String, Object>();
+        Map<String, Object> state = new HashMap<>();
```
Comment (ftomassetti) [inline javaparser-testing/src/test/java/com/github/javaparser/bdd/ManipulationTest.java:18]: Isn't he diamond operator Java7+?

File: javaparser-testing/src/test/java/com/github/javaparser/bdd/ParsingTest.java [hunk 4]
```diff
@@ -15,7 +15,7 @@
     
     @Override
     public InjectableStepsFactory stepsFactory() {
-        Map<String, Object> state = new HashMap<String, Object>();
+        Map<String, Object> state = new HashMap<>();
```
Comment (ftomassetti) [inline javaparser-testing/src/test/java/com/github/javaparser/bdd/ParsingTest.java:18]: ditto

File: pom.xml [hunk 5]
```diff
@@ -42,7 +42,7 @@
 
 	<properties>
 		<project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
-		<java.version>1.6</java.version>
+		<java.version>1.8</java.version>
```
Comment (ftomassetti) [inline pom.xml:45]: Java 8?

Comment (matozoid) [inline pom.xml:45]: That's the point: you can write everyting in Java 8, except for the core library, which is compiled with Java 6, and double checked to be compatible.

Comment (ftomassetti) [inline pom.xml:45]: I see, I missed the point :D

Comment (matozoid) [inline pom.xml:45]: Feel free to double check that everything compiles, and that the output library has the correct version. Maybe try adding a Java 8-only dependency to core, or add some Java 7 code, and see if the build fails as it should.

File: javaparser-testing/src/test/java/com/github/javaparser/bdd/visitors/PositionTestVisitor.java [hunk 6]
```diff
@@ -10,6 +10,8 @@
 import com.github.javaparser.ast.type.*;
 import com.github.javaparser.ast.visitor.VoidVisitorAdapter;
 
+import java.util.HashSet;
```
Comment (sebastiankirsch) [inline javaparser-testing/src/test/java/com/github/javaparser/bdd/visitors/PositionTestVisitor.java]: Is that intentional?

Comment (matozoid) [inline javaparser-testing/src/test/java/com/github/javaparser/bdd/visitors/PositionTestVisitor.java]: No, it's accidental.

**Comment by sebastiankirsch** [comment]
I'd give that a go, but I guess the travis.yml file causes a conflict?

**Comment by matozoid** [comment]
It's mergable again, who wants to?

**Pull request merged**